Document.cs

```diff
@@ -74,36 +72,68 @@ public class Document : IDisposable
 
 		#region Properties
 
-		DateTime? _ReleaseDate;
-		static readonly object releaseDateLock = new object();
+		private int _majorVersion;
+		private int _minorVersion;
+		private int _pointVersion;
+
 		/// <summary>
-		/// Gets the release date of the underlying Tidy library.
+		/// Returns the major version of libtidy.dll
 		/// </summary>
-		public DateTime ReleaseDate
+		public int MajorVersion
 		{
 			get
 			{
-				lock (releaseDateLock)
-				{
-					if (!_ReleaseDate.HasValue)
-					{
-						DateTime val = DateTime.MinValue;
-						string release = Marshal.PtrToStringAnsi(PInvoke.tidyReleaseDate());
-						if (release != null)
-						{
-							string[] tokens = release.Split(new char[] {' '}, StringSplitOptions.RemoveEmptyEntries);
-							if (tokens.Length >= 3)
-							{
-								DateTime.TryParseExact(tokens[0] + " " + tokens[1] + " " + tokens[2], "d MMMM yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out val);
-							}
-						}
-						_ReleaseDate = val;
-					}
-					return _ReleaseDate.Value;
-				}
+				if (_majorVersion != 0) return _majorVersion;
+
+				GetLibraryVersion();
+				return _majorVersion;
+			}
+		}
+
+		/// <summary>
+		/// Returns the minor version of libtidy.dll
+		/// </summary>
+		public int MinorVersion
+		{
+			get
+			{
+				if (_majorVersion != 0) return _minorVersion;
```

Comment:
This is correct but it looks suspicious. How about just taking out this test in all calling methods and putting it at the start of GetLibraryVersion?